src/views/Farms/components/BCakeMigrateModal.tsx

```diff
@@ -0,0 +1,218 @@
+    margin-right: calc((100% - 60px) / 2);
+    &::before {
+      position: absolute;
+      content: '';
+      width: calc(((100vw / 2) - 94px));
+      height: 2px;
+      top: 9px;
+      left: 30px;
+      transition: 0.3s background-color ease-in-out;
+      will-change: background-color;
+      background-color: ${({ theme, finished }) => (finished ? theme.colors.textSubtle : theme.colors.disabled)};
+    }
+    ${({ theme }) => theme.mediaQueries.md} {
+      &::before {
+        width: 90px;
+      }
+    }
+  }
+  ${StepperCircle} {
+    background: ${({ theme, finished }) => (finished ? theme.colors.textSubtle : theme.colors.disabled)};
+  }
+  ${StepperText} {
+    color: ${({ theme, active, finished }) =>
+      active ? theme.colors.primary : finished ? theme.colors.textSubtle : theme.colors.disabled};
+  }
+`
+
+export const FooterBox = styled.div`
+  margin-top: 24px;
+  padding-top: 24px;
+  position: relative;
+  &::before {
+    content: '';
+    position: absolute;
+    height: 1px;
+    width: 100%;
+    top: 0px;
+    left: 0px;
+    background-color: ${({ theme }) => theme.colors.cardBorder};
+  }
+`
+
+export const InfoBox = styled.div`
+  padding: 16px;
+  background: ${({ theme }) => theme.colors.background};
+  border: 1px solid ${({ theme }) => theme.colors.cardBorder};
+  border-radius: 16px;
+  font-size: 12px;
+  color: ${({ theme }) => theme.colors.textSubtle};
+  line-height: 120%;
+  width: 370px;
+  margin-bottom: 24px;
+`
+interface BCakeMigrateModalProps {
+  lpContract: Contract
+  stakedBalance: BigNumber
+  onUnStack: (amount: string, callback: () => void) => void
+  onDismiss?: () => void
+  pid: number
+  onUpdateFarm: () => void
+}
+
+type Steps = 'unStake' | 'enable' | 'stake'
+
+const migrationSteps: Record<Steps, string> = {
+  unStake: 'Unstake LP tokens from the farm',
+  enable: 'Enable staking with yield booster',
+  stake: 'Stake LP tokens back to the farm',
+}
+const migrationStepsKeys = Object.keys(migrationSteps)
+
+export const BCakeMigrateModal: React.FC<BCakeMigrateModalProps> = ({
+  lpContract,
+  stakedBalance,
+  onDismiss,
+  onUnStack,
+  pid,
+  onUpdateFarm,
+}) => {
+  const [activatedState, setActivatedState] = useState<Steps>('unStake')
+  const [isLoading, setIsLoading] = useState(false)
+  const [isApproved, setIsApproved] = useState(false)
+  const { t } = useTranslation()
+  const fullBalance = useMemo(() => {
+    return getFullDisplayBalance(stakedBalance)
+  }, [stakedBalance])
+  const { proxyAddress } = useBCakeProxyContractAddress()
+  const { onApprove } = useApproveBoostProxyFarm(lpContract, proxyAddress)
+  const bCakeProxy = useBCakeProxyContract(proxyAddress)
+  const { fetchWithCatchTxError, loading } = useCatchTxError()
+  const { toastSuccess } = useToast()
+
+  useEffect(() => {
+    if (!bCakeProxy) return
+    bCakeProxy.lpApproved(lpContract.address).then((enabled) => {
+      setIsApproved(enabled)
+    })
+  }, [lpContract, bCakeProxy])
```

Comment:
can move this effect into useApproveBoostProxyFarm, and use swr instead of effect